src/format.rs

```diff
@@ -0,0 +1,194 @@
+//! Contains abstractions for different kinds of
+//! serialization formats.
+//!
+//! Currently, the only two formats are [`Compact`] and [`Json`].
+
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::{fmt, str::FromStr};
+
+use base64ct::{Base64UrlUnpadded, Encoding};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::Unverified;
+
+pub(crate) mod sealed {
+    pub trait Sealed {}
+}
+
+/// Internal trait.
+pub trait FromFormat<F>: Sized + sealed::Sealed {
+    /// The error that can occurr while parsing `Self` from the input.
+    type Error;
+
+    /// Parse the input into a new instance of `Self`.
+    #[doc(hidden)]
+    fn from_format(input: F) -> Result<Unverified<Self>, Self::Error>;
+}
+
+/// Internal trait.
+#[doc(hidden)]
+pub trait IntoFormat<F>: sealed::Sealed {
+    fn into_format(self) -> F;
+}
+
+/// Internal trait.
+#[doc(hidden)]
+pub trait AppendSignatureTo<F>: sealed::Sealed {
+    fn append_to(self, f: &mut F);
+}
+
+impl<T: AsRef<[u8]>> sealed::Sealed for T {}
```

Comment:
Doesn't this implement `Sealed` for every `T: AsRef<[u8]>` including types outside this crate?